File: mobile/library/kotlin/io/envoyproxy/envoymobile/stats/Counter.kt
package io.envoyproxy.envoymobile

/** A time series counter. */
interface Counter {

  /** Increments the counter by the given count. */
  fun increment(count: Int = 1)

  /** Increments the counter by the given count and tags. */
  fun increment(tags: Tags = TagsBuilder().build(), count: Int = 1)
}

File: mobile/library/swift/stats/Counter.swift
import Foundation

/// A time series counter.
@objc
public protocol Counter: AnyObject {
  /// Increment the counter by the given count.
  ///
  /// - parameter count: The count to add to this counter.
  func increment(count: Int)

  /// Increment the counter by the given count and along with tags.
  ///
  /// - parameter tags:  The tags to attach to this Counter.
  /// - parameter count: The count to add to this counter.
  func increment(tags: Tags, count: Int)
}

extension Counter {
  /// Increment the counter by 1.
  public func increment() {
    self.increment(count: 1)
  }
}
